### ISSUE TITLE
Restricting access for not logged in users

### ISSUE THREAD

**Issue by robertpustulka** [cakephp/authorization#25]
Do we wan't to support some replacement for the `AuthComponent:allow()`/`deny()` methods?

Currently `AuthorizeComponent` requires that the identity is present in a request. There's no build-in way of checking if an identity is present and take some action if it is not (ie redirect to login page or throw a forbidden exception).

Not sure if this feature should belong here or maybe Authentication plugin?

**Comment by markstory** [comment]
How would you want that to be expressed in the controller action? Would something like 

```php
// Throw exception when no user is present
$this->Authorization->requireUserPresent()
```
work?

**Comment by robertpustulka** [comment]
I think that the current core Auth API is OK. We could port that to the plugin.

```php
// UsersController.php
public function initialize()
{
    ...
    $this->Authorization->deny('*');
    $this->Authorization->allow('login');
}
```

**Comment by robertpustulka** [comment]
Implemented by #30
